pyvista/core/_typing_core.py

```diff
@@ -1,13 +1,26 @@
 """Type aliases for type hints."""
-from typing import List, Sequence, Tuple, Union
+from typing import Sequence, Tuple, Union
 
 import numpy as np
+import numpy.typing as npt
 
 from pyvista.core._vtk_core import vtkMatrix3x3, vtkMatrix4x4, vtkTransform
 
-Vector = Union[List[float], Tuple[float, float, float], np.ndarray]
-VectorArray = Union[np.ndarray, Sequence[Vector]]
-Number = Union[float, int, np.number]
-NumericArray = Union[Sequence[Number], np.ndarray]
+NumpyNumArray = npt.NDArray[np.number]
```

Comment:
Unless we want to pin numpy version >=1.21, we cannot use `NDArray`. See https://numpy.org/doc/stable/reference/typing.html#numpy.typing.NDArray

Reply:
That will have to be for everyone to decide, but I'd be in favor of bumping it.

Reply:
I'm +1 to pin to >=1.21.